apps/meteor/server/models/raw/Messages.ts

```diff
@@ -1351,8 +1351,23 @@ export class MessagesRaw extends BaseRaw<IMessage> implements IMessagesModel {
 			query.tcount = { $exists: false };
 		}
 
+		const notCountedMessages = (
+			await this.find(
+				{
+					...query,
+					$or: [{ _hidden: true }, { editedAt: { $exists: true }, editedBy: { $exists: true }, t: 'rm' }],
+				},
+				{
+					projection: {
+						_id: 1,
+					},
+					limit,
+				},
+			).toArray()
+		).length;
+
 		if (!limit) {
-			const count = (await this.deleteMany(query)).deletedCount;
+			const count = (await this.deleteMany(query)).deletedCount - notCountedMessages;
```

Comment:
Since `limit` is `undefined` in this case, aren't the results not `limit`ed? :thinking: 

We can't just set it to `0` since there may be a `ts` filter in the query too. So we're not necessarily deleting _everything_ (all messages) within this method, it could also be all messages in a specific time period (or all messages except from threads, discussions etc.).